src/NHSD.BuyingCatalogue.Ordering.Api/Services/CreateOrderItem/OrderItemValidator.cs

```diff
@@ -0,0 +1,130 @@
+﻿using System;
+using System.Collections.Generic;
+using NHSD.BuyingCatalogue.Ordering.Api.Models;
+using NHSD.BuyingCatalogue.Ordering.Api.Services.UpdateOrderItem;
+using NHSD.BuyingCatalogue.Ordering.Api.Settings;
+using NHSD.BuyingCatalogue.Ordering.Domain;
+
+namespace NHSD.BuyingCatalogue.Ordering.Api.Services.CreateOrderItem
+{
+    public sealed class OrderItemValidator : ICreateOrderItemValidator, IUpdateOrderItemValidator
+    {
+        private readonly ValidationSettings _validationSettings;
+
+        public OrderItemValidator(ValidationSettings validationSettings)
+        {
+            _validationSettings = validationSettings ?? throw new ArgumentNullException(nameof(validationSettings));
+        }
+
+        public IEnumerable<ErrorDetails> Validate(CreateOrderItemRequest request)
+        {
+            if (request == null)
+                throw new ArgumentNullException(nameof(request));
+            if (request.Order == null)
```

Comment:
An improvement here I think would be to move the throw new ArgumentNullException's into the constructor of the `CreateOrderItemRequest`, that way by design the Validator wouldn't need to worry about it & any other use of the request can be sure the Order property is not null, but this could be done at a later date if you see fit.

Reply:
That would be nice, I wanted to ensure that this function had everything it needed, as that stops resharper from complaining about nulls not being checked. Will consider it during the rework.